src/daemon/Display.cpp

```diff
@@ -113,21 +115,30 @@ namespace SDDM {
         m_socketServer(new SocketServer(this)),
         m_greeter(new Greeter(this))
     {
+
         // Create display server
         switch (m_displayServerType) {
         case X11DisplayServerType:
+#ifdef BUILD_WITH_X11
             if (seat()->canTTY()) {
                 m_terminalId = VirtualTerminal::setUpNewVt();
             }
             m_displayServer = new XorgDisplayServer(this);
             break;
+#endif
```

Comment:
IMO this should be merged for easier readability:

```
        switch (m_displayServerType) {
#if BUILD_WITH_X11
        case X11DisplayServerType:
            if (seat()->canTTY()) {
                m_terminalId = VirtualTerminal::setUpNewVt();
            }
            m_displayServer = new XorgDisplayServer(this);
            break;
        case X11UserDisplayServerType:
        ....
#else
        case X11DisplayServerType:
        case X11UserDisplayServerType:
        ...
        [[fallthrough]];
#endif
        case WaylandDisplayServerType:
```

Reply:
Fixed.